include/powsybl/iidm/Network.hpp

```diff
@@ -150,6 +150,19 @@ class Network : public Container, public VariantManagerHolder {
 
     const stdcxx::DateTime& getCaseDate() const;
 
+    template <typename T, typename = typename std::enable_if<std::is_base_of<Connectable, T>::value>::type>
+    unsigned long getConnectableCount() const;
+
+    stdcxx::const_range<Connectable> getConnectables() const;
+
+    stdcxx::range<Connectable> getConnectables();
+
+    template <typename T, typename = typename std::enable_if<std::is_base_of<Connectable, T>::value>::type>
+    stdcxx::const_range<T> getConnectables() const;
+
+    template <typename T, typename = typename std::enable_if<std::is_base_of<Connectable, T>::value>::type>
+    stdcxx::range<T> getConnectables();
```

Comment:
Is it possible to set a default parameter to T to avoid multiple overload (even if you specialize the code)

Reply:
Yes it works with a default template param => fixed

Reply:
Let's decide if optimize the index by specializing `getAll` for leaf objects...